src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java

```diff
@@ -461,10 +462,18 @@ void updatePrimaryCorrespondent(UUID caseUUID, UUID stageUUID, UUID primaryCorre
         log.info("Updated Primary Correspondent for Case: {} Correspondent: {}", caseUUID, primaryCorrespondentUUID, value(EVENT, PRIMARY_CORRESPONDENT_UPDATED));
     }
 
-    void updatePrimaryTopic(UUID caseUUID, UUID stageUUID, UUID primaryTopicUUID) {
+    void updatePrimaryTopic(UUID caseUUID, UUID stageUUID, UUID primaryTopicUUID, UUID textUUID) {
         log.debug("Updating Primary Topic for Case: {} Topic: {}", caseUUID, primaryTopicUUID);
         CaseData caseData = getCaseData(caseUUID);
         caseData.setPrimaryTopicUUID(primaryTopicUUID);
+        // If we have been passed a text UUID we should change the data to replace it in data with the topic UUID
+        // to allow
+        if (textUUID != null){
```

Comment:
I think rather than adding textUUID and checking for null on the existing method, we should create a new method that takes a textUUID. That way the original methods don't need to pass null.